spec-main/extensions-spec.ts

```diff
@@ -175,9 +190,98 @@ describe('chrome extensions', () => {
     });
   });
 
+  describe('chrome.webRequest', () => {
+    async function fetch (contents: WebContents, url: string) {
+      return contents.executeJavaScript(`fetch("${url}")`);
+    }
+
+    describe('onBeforeRequest', () => {
+      it('can cancel http requests', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.webContents.once('dom-ready', async () => {
+          await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+
+          await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
+        });
+
+        w.loadURL(url);
+      });
+
+      it('does not cancel http requests', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.webContents.once('dom-ready', async () => {
+          await expect(fetch(w.webContents, url)).to.have.property('status', 200);
+        });
+
+        w.loadURL(url);
+      });
+    });
+
+    it('takes precedence over Electron webRequest - http', async () => {
+      const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+      const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+      w.webContents.once('dom-ready', async () => {
+        await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+
+        customSession.webRequest.onBeforeRequest(() => {
+          throw new Error('Electron onBeforeRequest callback has been called');
+        });
+
+        await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
+      });
+
+      w.loadURL(url);
+    });
+
+    it('takes precedence over Electron webRequest - WebSocket', async () => {
```

Comment:
Should this test that it only takes precedence on urls that the extension specifically intercepts?

e.g. if an extension registers webRequest handlers for `*.google.com`, and the main process registers handlers through the Electron API for `*.slack.com`, both handlers should work.

Reply:
Is that okay?
https://github.com/sentialx/electron/blob/a1cdc70435d953e0ad45fb8316b08dabbadc053d/spec-main/extensions-spec.ts#L232-L242

Reply:
I'm a little confused how that test works. How does the `onBeforeRequest` handler get invoked given that it's supposed to only trigger on google.com urls, which aren't ever requested afaict?

Reply:
Ah yes, I forgot, sorry.